engine/src/main/java/org/terasology/utilities/procedural/SubSampledNoise.java

```diff
@@ -65,13 +67,41 @@ public float noise(float x, float y) {
         return TeraMath.biLerp(q00, q10, q01, q11, xMod / sampleRate, yMod / sampleRate);
     }
 
+    /**
+     *
+     * @param region
+     * @return
+     * @deprecated This is scheduled for removal in an upcoming version method will be replaced with JOML implementation
+     *     {@link #noise(BlockArea)}.
+     */
+    @Deprecated
     public float[] noise(Rect2i region) {
         Rect2i fullRegion = determineRequiredRegion(region);
         float[] keyData = getKeyValues(fullRegion);
         float[] fullData = mapExpand(keyData, fullRegion);
         return getSubset(fullData, fullRegion, region);
     }
 
+    public float[] noise(BlockArea region) {
```

Comment:
As this does not modify the `region` if I see it correctly, this should be 
```suggestion
    public float[] noise(BlockAreac area) {
```

Also, let's try to call all `BlockArea`s `area` instead of `region` to avoid confusion with `BlockRegion`.